### ISSUE TITLE
Pulled/Unpulled not shown when upstream branch is local

### ISSUE THREAD

**Issue by a-m-s** [tpope/vim-fugitive#2185]
The `:Git` status page show Pulled/Unpulled status when `branch.<name>.remote` points to a real remote, but they're absent when the `remote = .` (i.e. "upstream" is a local branch).

The following snippet, from fugitive.vim, has handling for when `fetch_remote ==# '.'`, but that can never happen because `empty(config.Get('remote.' . fetch_remote . '.fetch'))` is always true.

```vim
let fetch_remote = config.Get('branch.' . branch . '.remote', 'origin')
let push_remote = config.Get('branch.' . branch . '.pushRemote',
      \ config.Get('remote.pushDefault', fetch_remote))
if empty(config.Get('remote.' . fetch_remote . '.fetch'))
  let fetch_remote = ''
endif
if empty(config.Get('remote.' . push_remote . '.push', config.Get('remote.' . push_remote . '.fetch')))
  let push_remote = ''
endif

let pull_type = 'Pull'
if empty(fetch_remote) || empty(branch)
  let pull_ref = ''
elseif fetch_remote ==# '.'
  let pull_ref = config.Get('branch.' . branch . '.merge', 'refs/heads/' . branch)
else
  let pull_ref = substitute(config.Get('branch.' . branch . '.merge', 'refs/heads/' . branch), '^refs/heads/', 'refs/remotes/' . fetch_remote . '/', '')
endif
```

The obvious patch appears to fix it, although I didn't do extensive testing yet.

```patch
--- a/autoload/fugitive.vim
+++ b/autoload/fugitive.vim
@@ -2806,7 +2806,7 @@ function! fugitive#BufReadStatus(...) abort
     let fetch_remote = config.Get('branch.' . branch . '.remote', 'origin')
     let push_remote = config.Get('branch.' . branch . '.pushRemote',
           \ config.Get('remote.pushDefault', fetch_remote))
-    if empty(config.Get('remote.' . fetch_remote . '.fetch'))
+    if fetch_remote !=# '.' && empty(config.Get('remote.' . fetch_remote . '.fetch'))
       let fetch_remote = ''
     endif
     if empty(config.Get('remote.' . push_remote . '.push', config.Get('remote.' . push_remote . '.fetch')))
```

**Comment by a-m-s** [comment]
That patch does the wrong thing when no upstream branch is set.

**Comment by tpope** [comment]
You're out of date. That was fixed in #2137.
